packages/portfolio-contract/src/portfolio.exo.ts

```diff
@@ -144,6 +146,62 @@ const eventAbbr = (e: VTransferIBCEvent) => {
   return { destination_channel: dest, sequence };
 };
 
+// Interface definitions for PortfolioKit facets
+const TapI = M.interface('PortfolioTap', {
+  receiveUpcall: M.call(M.record()).returns(M.promise()),
+});
+
+const ParseInboundTransferWatcherI = M.interface('ParseInboundTransferWatcher', {
+  onRejected: M.call(M.any()).returns(),
+  onFulfilled: M.call(M.any()).returns(M.promise()),
+});
+
+const ReaderI = M.interface('PortfolioReader', {
+  getLocalAccount: M.call().returns(M.remotable('LocalAccount')),
+  getStoragePath: M.call().returns(VowShape),
+  getPortfolioId: M.call().returns(M.number()),
+  getGMPInfo: M.call(M.string()).returns(VowShape),
+  getTargetAllocation: M.call().returns(M.or(TargetAllocationShapeExt, M.undefined())),
+});
+
+const ReporterI = M.interface('PortfolioReporter', {
+  publishStatus: M.call().returns(),
+  allocateFlowId: M.call().returns(M.number()),
+  publishFlowSteps: M.call(M.number(), M.arrayOf(M.record())).returns(),
+  publishFlowStatus: M.call(M.number(), M.record()).returns(),
+});
+
+const ManagerI = M.interface('PortfolioManager', {
+  reserveAccount: M.call(M.string()).returns(M.or(VowShape, M.undefined())),
+  resolveAccount: M.call(M.record()).returns(),
+  releaseAccount: M.call(M.string(), M.any()).returns(),
+  providePosition: M.call(PoolKeyShapeExt, M.string(), M.string()).returns(M.remotable('Position')),
+  waitKLUDGE: M.call(M.bigint()).returns(VowShape),
+  setTargetAllocation: M.call(TargetAllocationShapeExt).returns(),
+});
+
+const AccountWatcherI = M.interface('AccountWatcher', {
+  onRejected: M.call(M.any(), M.string()).returns(),
+});
+
+const RebalanceHandlerI = M.interface('RebalanceHandler', {
+  handle: M.call(M.remotable('ZCFSeat'), M.any()).returns(M.promise()),
```

Comment:
add a comment about how the 2nd arg, `offerArgs`, is checked against `offerArgsShapes.rebalance` within the method. We can't make that shape here because it includes a brand that's not available at module initialization time.

We could postpone definition of this interface until the body of `prepare...`, at which point we have `offerArgsShapes.rebalance`.